pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -856,6 +862,12 @@ func (l *LibvirtDomainManager) SyncVMI(vmi *v1.VirtualMachineInstance, allowEmul
 			if err != nil {
 				return nil, err
 			}
+			// Cache the metadata which has been updated by the converter.
+			// TODO: Perform this directly and not through the converter.
+			l.metadataCache.UID.Store(vmi.UID)
+			l.metadataCache.GracePeriod.Store(
+				api.GracePeriodMetadata{DeletionGracePeriodSeconds: converter.GracePeriod(vmi)},
+			)
```

Comment:
Shouldn't this be occurring before the domain is defined (a few lines above this)?